File: 728.cpp
class Solution {
public:
    vector<int> selfDividingNumbers(int left, int right) {
        vector<int> res;
        int num, digit;
        bool div;

        for(int i = left; i <= right; i++) {
            num = i;
            div = true;
            while(num != 0) {
                digit = num % 10;
                num /= 10;
                if(digit == 0) {
                    div = false;
                    break;
                }
                if(i % digit != 0) {
                    div = false;
                    break;
                }
            }
            if(div) {
                res.push_back(i);
            }
        }
        return res;
    }
};

File: 728.py
class Solution:
    def selfDividingNumbers(self, left: int, right: int) -> List[int]:
        res = []
        for i in range(left, right + 1):
            x = i
            dividing = True
            while(x != 0):
                digit = x % 10
                x //= 10
                if((digit == 0) or ((i % digit) != 0)):
                    dividing = False
                    break
            if(dividing):
                res.append(i)
        return res